### PR TITLE
Add support for Azure ZenServer

### PR DESCRIPTION
**Pull request by wjayesh**
## Describe changes
I implemented:
- an Azure ZenServer recipe
- an Azure ZenServer provider
- some fixes for AWS and GCP recipes
- support for init image tags

I still have some linting errors to fix.
## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/recipes/stack_recipe_service.py [hunk 1]
```diff
@@ -166,3 +179,52 @@ def get_service(cls, recipe_path: str) -> Optional["StackRecipeService"]:
...
+            file_path=os.path.join(
+                self.terraform_client.working_dir, "metadata.yaml"
+            )
+        )["Cloud"]
+
+        config = {
+            "name": f"{provider}",
+            "provider": f"{provider}",
+            "username": "default",
+            "password": "zenml",
```
Comment (stefannica) [inline src/zenml/recipes/stack_recipe_service.py:217]: Just an idea: if you leave out the username and password, the user will be prompted to enter them by the `zenml deploy` CLI command execution. Isn't that better than using default values here ?

File: src/zenml/zen_server/deploy/terraform/recipes/azure/printf.cmd [hunk 2]
```diff
@@ -0,0 +1,2 @@
+@echo off
+echo {"os": "Windows"}
```
Comment (stefannica) [inline src/zenml/zen_server/deploy/terraform/recipes/azure/printf.cmd:2]: Is this really used anywhere ? And if yes, why is hard-coded to windows ?

**Review by stefannica — APPROVED** [review]
Really nice to see Azure in the list of providers :) I found a few places where the code could be slightly improved, but nothing that can't be done at a later time.

File: src/zenml/cli/stack_recipes.py [hunk 3]
```diff
@@ -905,45 +959,51 @@ def deploy(
...
+def zen_server_exists() -> bool:
+    """Check if a remote ZenServer is active.
+
+    Returns:
+        True if active, false otherwise.
+    """
+
+    return not GlobalConfiguration().zen_store.is_local_store()
+
+
```
[the remark this answers is not in the document]
Comment (wjayesh) [reply src/zenml/cli/stack_recipes.py:1006]: Yes, the intention is to check if there is any remote server that the user is connected to.

File: src/zenml/recipes/stack_recipe_service.py (hunk 1 above)
Comment (wjayesh) [reply src/zenml/recipes/stack_recipe_service.py:217]: that's certainly better. Thanks for the idea!

**Pull request merged**